src/main/java/com/google/sps/servlets/ScheduledInterviewServlet.java

```diff
@@ -95,4 +112,62 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
   }
+
+  public List<ScheduledInterviewRequest> objectToRequestObject(
+      List<ScheduledInterview> scheduledInterviews, String timeZoneIdString) {
+    ZoneId timeZoneId;
+    if (timeZoneIdString == null) {
+      timeZoneId = ZoneId.systemDefault();
+    } else {
+      timeZoneId = ZoneId.of(timeZoneIdString);
+    }
+    List<ScheduledInterviewRequest> requestObjects = new ArrayList<ScheduledInterviewRequest>();
+    for (ScheduledInterview scheduledInterview : scheduledInterviews) {
+      requestObjects.add(makeSheduledInterviewRequest(scheduledInterview, timeZoneId));
+    }
+    return requestObjects;
+  }
+
+  private String getDateString(TimeRange when, ZoneId timeZoneId) {
+    LocalDateTime start = LocalDateTime.ofInstant(when.start(), timeZoneId);
+    LocalDateTime end = LocalDateTime.ofInstant(when.end(), timeZoneId);
+    String startTime = start.format(DateTimeFormatter.ofPattern("h:mm a"));
+    String endTime = end.format(DateTimeFormatter.ofPattern("h:mm a"));
+    String day = start.format(DateTimeFormatter.ofPattern("EEEE, MMMM d, yyyy"));
+    return day + " from " + startTime + " to " + endTime;
+  }
+
+  private ScheduledInterviewRequest makeSheduledInterviewRequest(
+      ScheduledInterview scheduledInterview, ZoneId timeZoneId) {
+    String userEmail = UserServiceFactory.getUserService().getCurrentUser().getEmail();
+    String userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
+    if (userId == null) {
+      userId = String.format("%d", userEmail.hashCode());
+    }
+    String date = getDateString(scheduledInterview.when(), timeZoneId);
+    String interviewer;
+    String interviewee;
+    String role;
+
+    if (userId.equals(scheduledInterview.interviewerId())) {
+      role = "Interviewer";
+    } else {
+      role = "Interviewee";
+    }
+
+    if (!personDao.get(scheduledInterview.interviewerId()).isPresent()) {
+      interviewer = "Nonexistent User";
```

Comment:
I'm still not sure the "Nonexistent User" is a good option. We're generally assuming that interviews will only be scheduled with existing Persons and that a Person is never deleted.

Reply:
I discussed this with Grant a bit.

It seems unclear what else we ought to do in this case. This function isn't really responsible for fixing instances of ScheduledInterview whose associated Person(s) aren't in the DB. The only choice really is to either (1) render something somewhat weird or (2) throw an exception we do not catch and return a 500.

(1) seems better to me.